### PR TITLE
`web/charts`: Migrate to null safety

### PR DESCRIPTION
**Pull request by domesticmouse**
## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: web/charts/pubspec.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ description: Charts-Flutter Demo
 publish_to: none
 
 environment:
-  sdk: '>=2.10.0 <3.0.0'
+  sdk: '>=2.12.0 <3.0.0'
```
Comment (johnpryan) [inline web/charts/pubspec.yaml:6]: nit: we're still fetching `charts_flutter` from GitHub - it looks like we could import it from Pub now?

File: web/charts/lib/app_config.dart [hunk 2]
```diff
@@ -17,10 +17,10 @@ import 'package:flutter/material.dart';
 
 /// A particular configuration of the app.
 class AppConfig {
-  final String appName;
-  final String appLink;
-  final ThemeData theme;
-  final bool showPerformanceOverlay;
+  final String? appName;
```
Comment (johnpryan) [inline web/charts/lib/app_config.dart:20]: I think these fields could be made non-nullable

File: web/charts/pubspec.yaml (hunk 1 above)
Comment (domesticmouse) [reply web/charts/pubspec.yaml:6]: The latest published version is broken w.r.t Flutter 2.5. Stuart Morgan is in the process of rolling up a release of `charts_flutter` that will enable me to revert this git dependency and then we will look at how to support the package going forward.

File: web/charts/lib/app_config.dart (hunk 2 above)
Comment (domesticmouse) [reply web/charts/lib/app_config.dart]: done

**Pull request merged**